packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/hooks/useReviewWorkflowsStages.js

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+import { useFetchClient } from '@strapi/helper-plugin';
+import { useQuery } from 'react-query';
+
+export function useReviewWorkflowsStages({ id, layout } = {}, queryOptions = {}) {
+  const { kind, uid } = layout;
+  const slug = kind === 'collectionType' ? 'collection-types' : 'single-types';
+
+  const { get } = useFetchClient();
+
+  const { data, isLoading } = useQuery(
+    ['content-manager', slug, layout.uid, id, 'stages'],
+    async () => {
+      const { data } = await get(`/admin/content-manager/${slug}/${uid}/${id}/stages`);
+
+      return data;
+    },
+    queryOptions
+  );
+
+  // these return values need to be memoized, because the default value
+  // would lead to infinite rendering loops when used in a dependency array
+  // on an effect
+  const meta = React.useMemo(() => data?.meta ?? {}, [data?.meta]);
+  const stages = React.useMemo(() => data?.data ?? [], [data?.data]);
```

Comment:
do these actually need memoizing? they'll be stable from the `useQuery` if they're present 🤔

Reply:
No - the default value of `{}` or `[]` makes them unstable.

Reply:
the comment is, i'm just challenging it is all 😛 should we be using `initialData` in this case then? It just feels overkill to use memo just for this 🤔

Reply:
Yes, I think using `initialData` is a good idea and I had that one too recently. This is the pattern I have introduced for all other data-fetching hooks so far and would like to keep it consistent for now here.

Will create a follow-up to make sure we do the same everywhere + extend your amazing FE guidelines on react-query once I am at it, ok?

Reply:
> Will create a follow-up to make sure we do the same everywhere + extend your amazing FE guidelines on react-query

lol okay ✅